src/utils/prim-lexing.fs

```diff
@@ -2,10 +2,96 @@
 
 #nowarn "47" // recursive initialization of LexBuffer
 
+namespace Microsoft.FSharp.Compiler.Text
+
+open System
+open System.IO
+
+type ISourceText =
+
+    abstract Item : int -> char with get
+
+    abstract GetLines : unit -> string []
```

Comment:
I've initially thought it would fit R# APIs better as there is APIs for getting a single line text and no method to get an array with all of them
I've just checked and in practice, though, it seems to only keep line start/end offsets and always creates substrings for lines with using any of current APIs.

Reply:
To be honest, I really want to get rid of `GetLines`. We only use it in xml parser service.